03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
-we might explore a dataset that we collected in the field. We can do this using
-the `hist` function in `R`.
-
-What range of values do you expect to see for the Harvard Forest `Canopy Height 
-Model (CHM)` that we just created?
-
-
-```{r view-histogram }
-
-#Create histogram of CHM values
+Notice the legend on the right.  We are now looking at the height of the canopy 
+instead of the elevation hence the smaller numbers.  
+
+#Challenge: Exploring Raster Values in CHM
+As we learned in previous lessons, it's often a good idea to explore the range 
+of values in a raster dataset just like we might explore a dataset that we 
+collected in the field.  
+
+1. What range of values do you expect to see for the Harvard Forest Canopy 
+Height Model (`CHM`) that we just created? 
+2. What are two ways you can check this range of data in CHM? 
+3. Plot your data in such a way that you can see the range of data values in 
+CHM.
+4. Bonus challenge: Change the color of and add a title to the plot you created 
+in question 3. 
+
+```{r challenge-code-1 }
+#1) Depends on types of trees.  0m to <50m?
+#2) Looks at histogram or min/max values. 
+minValue(CHM)
+maxValue(CHM)
+#3) 
+hist(CHM, maxpixels=ncell(CHM))
+
+#4)
 hist(CHM, 
      col = "purple",
-     main = "Histogram of NEON Canopy Height Model\n Harvard Forest")
+     main = "Histogram of NEON Canopy Height Model\n Harvard Forest",
+     maxpixels=ncell(CHM))
 
 ```
 
-#Raster Overlay
-
-Raster math is an appropriate aporoach to this calculation if
+#Efficient Raster Math: Overlay Function
+Basic raster math like we just did, is an appropriate aporoach to a calculation 
+if
 
 1. The rasters we are using are small in size.
 2. The calculations we are performing are simple.
 
-Raster math is easy to code up, however it is less efficient as computation becomes
-more complex or as file sizes become larger. To deal with larger rasters that are
-stored as individual files we should use the `overlay` function
+Raster math is intuitive and easy to code, however it is less efficient as 
+computation becomes more complex or as file sizes become larger. The `overlay()`
+function is more efficient when
 
-NOTE: If our rasters are stacked and stored as `RasterStack` or `RasterBrick`
-objects in R, then we can use `calc`.
-{: .notice }
+1. The rasters we are using are larger in size. 
+2. The rasters are stored as individual files. 
+3. The computations needed are complex. 
 
-Let's have a look at how the `overlay` function works. The `overlay` function 
-takes two or more rasters and applies a function to them using efficient processing
-methods. The syntax looks like:
+>NOTE: If the rasters are stacked and stored as `RasterStack` or `RasterBrick`
+>objects in R, then we should use `calc()`. `overlay()` will not work on stacked
+>rasters. 
 
-`outputRaster <- overlay(raster1, raster2, functionName )`
+The `overlay()` function takes two or more rasters and applies a function to
```

Comment:
What I get from this is I can use overlay() to apply a function to any two (or more) rasters.  The function needs to be specified.  I don't get an understanding of HOW is happens, just that it is more efficient (although not noticeably faster in our example dataset).